### ISSUE TITLE
Suites from multiple projects run in parallel even when restricting Tags.Test to 1

### ISSUE THREAD

**Issue by rabejens** [sbt/sbt#2850]
(See the guidelines for contributing, linked above)

I have a large multi-project SBT build. I have a `Settings.scala` which defines settings like these:

```scala
  lazy val commonSettings = Seq(
    fork in test := true,
    parallelExecution in test := false,
    fork in testQuick := true,
    parallelExecution in testQuick := false,
    // more
  )
```

Each project imports them like this:

```scala
lazy val some_project = project
  .in(file("some/project"))
  .settings(commonSettings)
```

Now, when I try to run `sbt test` the suites are still run in parallel. I read in another issue that I should add:

```scala
concurrentRestrictions in Global += Tags.limit(Tags.Test, 1)
```

But no matter where I add this, be it to `build.sbt` directly or to my `commonSettings`, it is ignored and the suites are run in parallel. I am on the verge of resorting to polling lock files to serialize my tests because I don't want them to run in parallel.

Is this a bug or am I doing anything wrong?

**Comment by rabejens** [comment]
I finally solved it.

It was a cascade of several issues. I now put the following in my main `build.sbt` file:

```scala
fork in test := true
parallelExecution in test := false
fork in testQuick := true
parallelExecution in testQuick := false
concurrentRestrictions in Global += Tags.limit(Tags.Test, 1)
```

The second thing was that I was using `cassandra-unit` to spin up a temporary Cassandra database for some of my tests, and each of my tests did

```scala
EmbeddedCassandraServerHelper.startEmbeddedCassandra()
```

So it tried to start multiple C* servers which fails as I expect it to. I first wrote a wrapper that first tries to connect to a local C* server and starts the cassandra-unit one only if that fails, but C* unit also has another flaw in that it slows my computer WAY down - even to the point that the mouse pointer begins to move choppily. So, I am now relying on a local C* server running before the test is started. When building this on my Jenkins, I am just adding a pre- and post-build action that spins up and kills the local C* server respectively.
